src/hooks/useLedgerConnect.ts

```diff
@@ -0,0 +1,230 @@
+import AppEth from '@ledgerhq/hw-app-eth';
+import TransportBLE from '@ledgerhq/react-native-hw-transport-ble';
+import { forEach } from 'lodash';
+import { useCallback, useEffect, useState } from 'react';
+import useWallets from './useWallets';
+import { getHardwareKey, getHdPath, WalletLibraryType } from '@/model/wallet';
+import { DebugContext } from '@/logger/debugContext';
+import { logger, RainbowError } from '@/logger';
+import * as i18n from '@/languages';
+
+enum LEDGER_ERROR_CODES {
+  OFF_OR_LOCKED = 'off_or_locked',
+  NO_ETH_APP = 'no_eth_app',
+  UNKNOWN = 'unknown',
+  DISCONNECTED = 'disconnected',
+}
+
+export const getLedgerErrorText = (errorCode: LEDGER_ERROR_CODES) => {
+  switch (errorCode) {
+    case LEDGER_ERROR_CODES.OFF_OR_LOCKED:
+      return i18n.t(i18n.l.hardware_wallets.errors.off_or_locked);
+    case LEDGER_ERROR_CODES.NO_ETH_APP:
+      return i18n.t(i18n.l.hardware_wallets.errors.no_eth_app);
+    default:
+      return i18n.t(i18n.l.hardware_wallets.errors.unknown);
```

Comment:
do we need one for disconnected state?

Reply:
not atm but may end up adding one, i wanna keep that one without an error state since if its disconnected we want to clear everything and start with a fresh connection and i dont want to flash an error state without actionable info